compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeL1Allocations.cpp

```diff
@@ -114,6 +94,17 @@ LogicalResult distributeLocalMemory(ModuleOp moduleOp) {
   if (failed(maybeIndVars)) return failure();
   const DenseSet<Value> &indVars = maybeIndVars.value();
   IRRewriter rewriter(moduleOp.getContext());
+
+  // Fetch all innermost scf.forall op.
+  DenseSet<Operation *> innermostForallOps;
+  moduleOp->walk([&](scf::ForallOp forallOp) {
+    bool isInnermost = true;
+    forallOp.walk([&](scf::ForallOp innerForallOp) {
+      if (innerForallOp != forallOp) isInnermost = false;
+    });
+    if (isInnermost) innermostForallOps.insert(forallOp);
+  });
```

Comment:
We should look for the `scf.forall` ops with thread mapping, because that's what we consider to be mapped to the cores.

Reply:
Done.